symfony/framework-bundle/3.3/public/index.php

```diff
@@ -24,6 +24,10 @@
     Debug::enable();
 }
 
+if ('cli-server' === PHP_SAPI && '/index.php' !== $_SERVER['SCRIPT_NAME'] && is_file(__DIR__.$_SERVER['SCRIPT_NAME'])) {
+    return false;
+}
```

Comment:
Not a big fan of adding something like this for all requests.

Reply:
Me neither, my first attempt was to put if inside the `if ($debug)` block. Would you prefer this?
I changed this behavior because sometimes it's convenient to use the built-in web server in "prod" mode to debug, but it can be considered an edge case.

Reply:
Yea, this block looks confusing... just to fix an edge case situation (where people should be using the web server bundle)

Reply:
Indeed, @weaverryan is right.

Reply:
> (where people should be using the web server bundle)

Doesn't it suffer from this same issue, since it's a wrapper around the PHP built-in web server?

Reply:
@weaverryan @fabpot in this case we should not explain how to use the built-in web server in the post install message and require web server bundle as a dev dependency of the skeleton, because it currently it’s very disappointing for the end user. Here is how I figured the problem:

1. During a hackathon, a  newcomer installs Symfony through composer, then API Platform using composer req api
2. It starts the built-in PHP server using the displayed post install command
3. It tries to generate a client app using the client generator of API Platform

It fails because the generator try to access the /docs.jsonld route, and this bug prevent it to work properly.

It’s happening with the default integration of API Platform with Symfony, but it can also happen in a lot of unrelated use cases, and it’s almost impossible to figure what ‘s going on for a newcomer.

@teohhanhui no it doesn’t, because it comes with a router script fixing this problem.

Reply:
> we should not explain how to use the built-in web server in the post install message and require web server bundle as a dev dependency of the skeleton

:+1:

Reply:
@dunglas I agree with you, let's remove the reco from the post install message and recommend to use the web server bundle instead.

Reply:
@fabpot done

Reply:
@dunglas and @fabpot i haven't seen this PR before adding my PR #475 .
Adding **public/index.php** ( in server command  `php -S 127.0.0.1:8000 -t public public/index.php` ) it works fine for me and we can keep how to use the built-in web server in the post install  .
it's a good solution ?

Reply:
Not it's not enough (see previous commits).